vip-jetpack/vip-jetpack.php

```diff
@@ -9,6 +9,34 @@
  * License: GPL2+
  */
 
+/** 
+ * Lowest incremental sync queue size allowed on VIP - JP default is 1000, but we're bumping to 10000 to give VIPs more
+ * headroom as they tend to publish more than average
+ */
+define( 'VIP_GO_JETPACK_SYNC_MAX_QUEUE_SIZE_LOWER_LIMIT', 10000 );
+
+/**
+ * The largest incremental sync queue size allowed - items will not get enqueued if there are already this many pending
+ * 
+ * The queue is stored in the option table, so if the queue gets _too_ large, site performance suffers
+ */
+define( 'VIP_GO_JETPACK_SYNC_MAX_QUEUE_SIZE_UPPER_LIMIT', 50000 );
+
+/**
+ * The lower bound for the incremental sync queue lag - if the oldest item has been sitting unsynced for this long,
+ * new items will not be added to the queue
+ * 
+ * The default is 15 minutes, but VIP sites often have more busy queues and we must prevent dropping items if the sync is
+ * running behind
+ */
+define( 'VIP_GO_JETPACK_SYNC_MAX_QUEUE_LAG_LOWER_LIMIT', 2 * HOUR_IN_SECONDS );
+
+/**
+ * The maximum incremental sync queue lag allowed - just sets a reasonable upper bound on this limit to prevent extremely
+ * stale incremental sync queues
+ */
+define( 'VIP_GO_JETPACK_SYNC_MAX_QUEUE_LAG_UPPER_LIMIT', 6 * HOUR_IN_SECONDS );
```

Comment:
@lezama / @mdbitz same question - is 6 hours too conservative for a max queue lag setting?

Reply:
I guess it depends what we have promised to our customers. On a healthy site the queue should never be bigger than a dozen actions. If they reach this numbers it is probably because something else is wrong, failing in wpcom or in the communication channel. We need to keep this big just for when there's a situation where we don't want to lose this actions while we fix things.

Reply:
It also depends on how much the size of the queue affects  the site.
We need to remember two things:
- The queue gets stored as blog options
- Both conditions have to be met in order to stop adding to the queue. So the queue could have more than `VIP_GO_JETPACK_SYNC_MAX_QUEUE_SIZE_UPPER_LIMIT` if the lag hasn't reached `VIP_GO_JETPACK_SYNC_MAX_QUEUE_LAG_UPPER_LIMIT`

Reply:
> Both conditions have to be met in order to stop adding to the queue

Ah - in that case, I may have made the limits _too high_ in https://github.com/Automattic/vip-go-mu-plugins/pull/1352